### PR TITLE
feat(profile): add organization pinning to user profiles + fix broken profile and home tests

### LINKED ISSUE #369: Add organization pinning on user profile
### **What**

Users can now pin **one organization** to their profile, making it visible directly on their profile header and profile picture. This includes UI support for pinning and unpinning organizations, persistent storage of the pinned organization, and visual indicators across profile-related screens.

### **Why**

Many users belong to multiple organizations, but currently have no way to highlight the most relevant one. This makes profiles less expressive and reduces organization visibility. Allowing users to pin an organization improves profile personalization and clearly communicates their primary affiliation.

### **How**

* Extend the `User` model with a `pinnedOrganizationId` field.
* Add pin, unpin, and fetch logic to user repositories (Firestore and local).
* Add pin/unpin actions in the organization management screen.
* Display the pinned organization as a badge on the profile picture.
* Load and refresh pinned organization state in profile and settings screens.
* Add comprehensive unit and UI tests to ensure correctness and stability.

### PR DESCRIPTION
**Pull request by othmanehousni**
> [!IMPORTANT]
> This PR is also a fix PR that fixes the tests that were wrongly merged between the main and the last PR
> PR documentation, description and tests were done by an AI for better readability and maintainability 

# **What**

This PR introduces a **pinning system for organizations on user profiles**, allowing users to pin a single organization that is prominently displayed on their profile. It adds full UI support for pinning and unpinning organizations, displays the pinned organization as a badge on the profile picture, and integrates the feature across profile, organization management, and settings screens. The change also includes comprehensive unit and UI tests to ensure correctness and stability.

# **Why**

Users who are members of organizations currently have no way to highlight their primary or most relevant organization on their profile. This makes profiles less expressive and reduces visibility for organizational involvement. Adding a pinning mechanism improves profile personalization, makes organizational affiliations clearer at a glance, and aligns the profile experience with pinned events already supported by the app.

# **How**

* **Data layer**

  * Added `pinnedOrganizationId` to the `User` model with Firestore and local repository support.
  * Extended `UserRepository` with `pinOrganization`, `unpinOrganization`, and `getPinnedOrganization`.
* **ViewModels**

  * Updated profile and organization management view models to load, pin, and unpin organizations.
  * Ensured pinned organization state is refreshed on resume and handled gracefully on errors.
* **UI**

  * Added pin/unpin controls in the organization management screen.
  * Displayed the pinned organization as a badge overlay on the profile picture, including logo support and a visual fallback.
  * Ensured consistent behavior across profile, profile settings, and visitor views.
* **Testing**

  * Added extensive unit tests for user models, repositories, and view models.
  * Added UI and integration tests covering pin/unpin behavior, edge cases, and error handling.

# Linked issue

Closes #369

image

image

image

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/model/media/MediaRepositoryFirebaseStorage.kt [hunk 1]
```diff
@@ -26,7 +26,25 @@ class MediaRepositoryFirebaseStorage(
...
+            storage.reference.child(decodedPath)
+          } else {
+            // Fallback to using id as-is
+            storage.reference.child(id)
+          }
+        } else {
+          // It's a direct storage path
+          storage.reference.child(id)
+        }
+
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/model/media/MediaRepositoryFirebaseStorage.kt]: nice fix

File: app/src/main/java/com/github/se/studentconnect/ui/profile/OrganizationManagementViewModel.kt [hunk 2]
```diff
@@ -80,4 +85,43 @@ class OrganizationManagementViewModel(
...
+          Log.d(TAG, "Unpinned organization: $organizationId")
+        } else {
+          // Pin
+          userRepository.pinOrganization(userId, organizationId)
+          _uiState.value = _uiState.value.copy(pinnedOrganizationId = organizationId)
+          Log.d(TAG, "Pinned organization: $organizationId")
+        }
+      } catch (e: Exception) {
+        Log.e(TAG, "Failed to toggle pin organization", e)
+        _uiState.value = _uiState.value.copy(error = "Failed to update pin: ${e.message}")
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/profile/OrganizationManagementViewModel.kt]: this string should be in strings.xml

File: app/src/main/java/com/github/se/studentconnect/ui/profile/components/ProfileHeader.kt [hunk 3]
```diff
@@ -171,37 +183,195 @@ fun ProfileHeader(
...
+  val textSizePx = with(density) { 8.sp.toPx() }
+
+  Box(modifier = modifier, contentAlignment = Alignment.Center) {
+    // Curved text above the badge
+    val orgName = organization.name
+
+    Box(
+        modifier =
+            Modifier.size(60.dp).drawBehind {
+              val radius = 25.dp.toPx()
```
Comment (83759) [inline app/src/main/java/com/github/se/studentconnect/ui/profile/components/ProfileHeader.kt]: nice job making sure there are no hardcoded .dp values!

**Review by 83759 — CHANGES_REQUESTED** [review]
Nice job, everything is documented, you made sure there are no hardcoded .dp values, most strings are in strings.xml, and line coverage is above 80%.

There are just 2 spots where you forgot to put the user-facing strings in strings.xml.
Fix that and this should be ready to merge!

File: app/src/main/java/com/github/se/studentconnect/ui/profile/OrganizationManagementViewModel.kt (hunk 2 above)
Comment (othmanehousni) [reply app/src/main/java/com/github/se/studentconnect/ui/profile/OrganizationManagementViewModel.kt]: done!

File: app/src/main/java/com/github/se/studentconnect/ui/screen/profile/OrganizationManagementScreen.kt [hunk 4]
```diff
@@ -398,37 +430,63 @@ private fun OrganizationCard(
...
+                            organization.memberUids.size),
                     style = MaterialTheme.typography.bodySmall,
                     color = MaterialTheme.colorScheme.onSurfaceVariant)
               }
+
+              // Pin button
+              IconButton(onClick = onPinClick) {
+                Icon(
+                    imageVector = if (isPinned) Icons.Filled.PushPin else Icons.Outlined.PushPin,
+                    contentDescription = if (isPinned) "Unpin organization" else "Pin organization",
```
[the remark this answers is not in the document]
Comment (othmanehousni) [reply app/src/main/java/com/github/se/studentconnect/ui/screen/profile/OrganizationManagementScreen.kt]: done

**Review by 83759 — APPROVED** [review]
thanks, you put the strings in strings.xml
